### PR TITLE
Close #134566: Added settings for terminal tab default color/icon

### LINKED ISSUE #134566: Add option to set the default color for the terminal on workspace folder
I often work with workspaces, when I open a new terminal it asks me on which directory I would like to start the terminal
image

It would be nice if in the `settings.json` located in the `.vscode` folder you could specify the default color (and names too) for terminals.

image

That would make terminals easier to distinguish, without having to change the colors and names every time.

### PR DESCRIPTION
**Pull request by amanasifkhalid**
This PR fixes #134566 by adding the settings `terminal.integrated.tabs.defaultColor` and `terminal.integrated.tabs.defaultIcon`. These settings support the same values as the "Change Terminal Color" and "Change Terminal Icon" actions. Below is a demo of this new workflow:

https://user-images.githubusercontent.com/58230338/163440980-e109eb5e-2efc-4e6a-896e-073c64633641.mp4

### CODE REVIEW

**Review by Tyriar — DISMISSED** [review]
Thanks for the PR @amanasifkhalid. This needed a little extra work, essentially replacing all instances of `Codicon.terminal` inside the terminal component as there's a bit of complexity to the "resolving" of the icons. I think I caught all the cases.

**Comment by Tyriar** [comment]
@meganrogge a full review would be helpful here as the change turned out to be fairly large

File: src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts [hunk 1]
```diff
@@ -135,15 +145,21 @@ export abstract class BaseTerminalProfileResolverService implements ITerminalPro
 
 		// Verify the icon is valid, and fallback correctly to the generic terminal id if there is
 		// an issue
-		shellLaunchConfig.icon = this._getCustomIcon(shellLaunchConfig.icon) || this._getCustomIcon(resolvedProfile.icon) || Codicon.terminal;
+		console.log('icon', this._configurationService.getValue(TerminalSettingId.TabsDefaultIcon));
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts:148]: remove

File: src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts [hunk 2]
```diff
@@ -135,15 +145,21 @@ export abstract class BaseTerminalProfileResolverService implements ITerminalPro
...
+		shellLaunchConfig.icon = this._getCustomIcon(shellLaunchConfig.icon)
+			|| this._getCustomIcon(resolvedProfile.icon)
+			|| this.getDefaultIcon();
+		console.log('icon2', shellLaunchConfig.icon);
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts:152]: remove

File: src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts (hunk 1 above)
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts:148]:
```suggestion
```

File: src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts [hunk 3]
```diff
@@ -135,15 +145,20 @@ export abstract class BaseTerminalProfileResolverService implements ITerminalPro
 
 		// Verify the icon is valid, and fallback correctly to the generic terminal id if there is
 		// an issue
-		shellLaunchConfig.icon = this._getCustomIcon(shellLaunchConfig.icon) || this._getCustomIcon(resolvedProfile.icon) || Codicon.terminal;
+		shellLaunchConfig.icon = this._getCustomIcon(shellLaunchConfig.icon)
+			|| this._getCustomIcon(resolvedProfile.icon)
+			|| this.getDefaultIcon();
+		console.log('icon2', shellLaunchConfig.icon);
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts:151]:
```suggestion
```

**Comment by meganrogge** [comment]
works well 👍🏼 . I wonder if we should add a tag or include icon in the setting title for the icon's color to make it more discoverable?

Screen Shot 2022-07-06 at 8 34 43 PM

**Comment by meganrogge** [comment]
Reset to default resets it to the default that's configured - I wonder if that's what a user would expect - I'd sort of expect the color to go back to grey
https://user-images.githubusercontent.com/29464607/177665586-b2c70482-3864-4258-80eb-a6ffd6d19ca6.mov

**Comment by meganrogge** [comment]
> works well 👍🏼 . I wonder if we should add a tag or include icon in the setting title for the icon's color to make it more discoverable?
> 
> Screen Shot 2022-07-06 at 8 34 43 PM

Going with `terminal.integrated.tabs.defaultIconId` and `terminal.integrated.tabs.defaultIconColor`

**Review by meganrogge — APPROVED** [review]

**Comment by Tyriar** [comment]
> Reset to default resets it to the default that's configured - I wonder if that's what a user would expect - I'd sort of expect the color to go back to grey

I had it doing this at some point and it didn't feel right. We can wait for feedback?

> Going with terminal.integrated.tabs.defaultIconId and terminal.integrated.tabs.defaultIconColor yields this and IMO is more accurate

I was trying to align the wording with profiles:

https://github.com/microsoft/vscode/blob/ce8b7a9c132b49a74d9ab7f4373ae57b3402755d/src/vs/platform/terminal/common/terminalPlatformConfiguration.ts#L27-L47

I'll revert your setting rename change since the profiles property is just `icon` and `color`, I created a follow up in https://github.com/microsoft/vscode/issues/154354 to add it to the setting description/any other polishing.

**Review by Tyriar — APPROVED** [review]

**Comment by meganrogge** [comment]
Just realized that this doesn't work in the case of the multi-root repository because the setting is per workspace ATM, so you can't distinguish between the terminals that way.

Folder settings are undefined when in a multi-root repo

**Comment by Tyriar** [comment]
> Folder settings are undefined when in a multi-root repo

Isn't it the other way; folder settings are only set when in a multi-root repo?

**Pull request merged**